llvm/include/llvm-c/ExternC.h

```diff
@@ -36,4 +36,18 @@
 #define LLVM_C_EXTERN_C_END LLVM_C_STRICT_PROTOTYPES_END
 #endif
 
+#ifndef LLVM_HAS_ATTRIBUTE
```

Comment:
Hmm, is this the best place for the code to live given the comment at the top of the file: `This file defines an 'extern "C"' wrapper` ?

Maybe this belongs in `Core.h`? or some new file for compiler-based macro definitions?

Reply:
Fair point - think I semi-randomly picked this as somewhere in llvm-c that was defining macros. But yeah, `ExternC.h` has a fairly clear purpose and this isn't related to that.

`Core.h` sounds OK - but also happy to start another header. It's sort of a lower-level (in the C API) version of `Compiler.h` - could pull some of `Compiler.h` down into this - but maybe not necessary to do that proactively (except maybe for the `LLVM_*_PREREQ` macros you mentioned - if we add a `LLVM_CLANG_PREREQ` macro down here, probably good to pull the others down so they're in the same place)